src/core/reducers/movies-reducer.ts

```diff
@@ -0,0 +1,83 @@
+import {ActionI, MovieI} from '../interface';
+
+export interface MoviesReducerStateI {
+    movies: Array<MovieI>;
+    selectedOption: string;
+    selectedGenre: string;
+    query: string;
+    sortOrder: string;
+    offset: number;
+    limit: number;
+    total: number | null;
+}
+
+export const moviesReducer = (state: MoviesReducerStateI = {
+    movies: [],
+    selectedOption: 'release_date',
+    selectedGenre: '',
+    query: '',
+    sortOrder: 'asc',
+    offset: 0,
+    limit: 6,
+    total: null
+}, action: ActionI) => {
+    switch (action.type) {
+        case "OPTIONS":
```

Comment:
Move strings to contants.
`const REQUESTS = { SET_OPTIONS: 'SET_OPTIONS' }`
Name as an action, not noun.